### PR TITLE
perltidy: init

### PR DESCRIPTION
**Pull request by haruki7049**
# Changes
- Init `/programs/perltidy.nix`
- Execute `/example.sh`, then it generates `/examples/formatter-perltidy.toml`

### CODE REVIEW

**Comment by haruki7049** [comment]
DONE

File: programs/perltidy.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    settings.formatter.perltidy = {
+      command = cfg.package;
+
+      options =
+        if cfg.perltidyrc == null then
+          cfg.options
+        else
+          [
+            "-pro=${cfg.perltidyrc}"
+          ] ++ cfg.options;
```
Comment (zimbatm) [inline programs/perltidy.nix:49]:
```suggestion
        lib.optional (cfg.perltidyrc != null) "-pro=${cfg.perltidyrc}"
        ++ cfg.options;
```
Something like that shows clearer intent

**Review by zimbatm — COMMENTED** [review]
Small nit, the rest looks good to me

**Comment by zimbatm** [comment]
thanks for the contrib!

**Pull request merged**